dotnet/Dockerfile

```diff
@@ -0,0 +1,11 @@
+FROM kubespheredev/builder-base:v3.1.0
+
+RUN rpm -Uvh https://packages.microsoft.com/config/centos/7/packages-microsoft-prod.rpm
+
+RUN yum install -y dotnet-sdk-5.0
```

Comment:
In fact, we can install both dotnet-sdk 5.0 and dotnet-sdk 3.1，to support building, packaging different version of dotnet apps without conflict in a single agent.  And u may wanna add 2.1 as well.  Such as
`Run yum install -y dotnet-sdk-5.0 dotnet-sdk-3.1 dotnet-sdk-2.1`
Besides, following the official time schedule, dotnet 6 will be released in November this year.  Now, the latest version is 6-Preview4. If new version of kubesphere dev-ops system will be released in next year, dotnet-sdk-6 could also be included .

Reply:
What about the size of the image if there're multiple version in one single image?

Reply:
> What about the size of the image if there're multiple version in one single image?

I built these images locally. Here is the result:

| Built with sdk(s) | Size |
| ------- | ---- |
| Only sdk-5.0 | 1.45GB |
| Only sdk-3.1 | 1.39GB |
| Only sdk-2.1 | 2.62GB |
| sdk-5.0 & sdk-3.1 | 1.77GB |
| sdk-5.0 & sdk-3.1 & sdk-2.1 | 3.35GB |

Reply:
In my view, `3.35G` is too big. I have a question about the version. There's many version for SDK, but I guess not all of them are popular for users. For example, Java8 and Java11 are very popular for Java developers. So, we might don't need to provide all the version.